### PR TITLE
record-editor: add and use GET endpoints that returns schema too

### PR DESCRIPTION
**Pull request by harunurhan**
Ref #1367

### CODE REVIEW

File: backend/tests/integration/editor/test_editor_views.py [hunk 1]
```diff
@@ -24,6 +24,24 @@
...
+        response = client.get(f"api/editor/conferences/{conference['control_number']}")
+
+    assert response.status_code == 200
+
+    response_data = json.loads(response.data)
+    record_metadata = response_data["record"]["metadata"]
+    schema = response_data["schema"]
+
+    assert record_metadata == dict(conference)
+    assert schema == load_schema("conferences")
```
Comment (drjova) [inline backend/tests/integration/editor/test_editor_views.py:42]: could we add an unhappy path test (no permissions) as well?

File: backend/inspirehep/editor/views.py [hunk 2]
```diff
@@ -44,6 +45,16 @@ def revert_to_revision(endpoint, pid_value):
         raise EditorRevertToRevisionError
 
 
+@blueprint.route("/<endpoint>/<int:pid_value>", methods=["GET"])
+@login_required_with_roles([Roles.cataloger.value])
+def get_record_and_schema(endpoint, pid_value):
+    pid_type = PidStoreBase.get_pid_type_from_endpoint(endpoint)
+    record = InspireRecord.get_record_by_pid_value(pid_value, pid_type)
+    return jsonify(
+        {"record": {"metadata": record}, "schema": load_schema(record["$schema"])}
```
Comment (drjova) [inline backend/inspirehep/editor/views.py:54]: we could also drop `metadata` right? Since we're changing the editor as well

Comment (harunurhan) [reply backend/inspirehep/editor/views.py:54]: I just kepth the this if we need to add more info for the record later, like updated etc.

File: backend/tests/integration/editor/test_editor_views.py [hunk 3]
```diff
@@ -24,6 +24,35 @@
...
+
+
+def test_get_record_and_schema_requires_cataloger_logged_in(inspire_app):
+    user = create_user()
+    conference = create_record("con")
+
+    with inspire_app.test_client() as client:
+        login_user_via_session(user, email=cataloger.email)
```
Comment (drjova) [inline backend/tests/integration/editor/test_editor_views.py:50]:
```suggestion
        login_user_via_session(user, email=user)
```

**Review by drjova — APPROVED** [review]
LGTM, just the typo in the test!!

**Pull request merged**